### PR TITLE
Feat/index genesis transactions

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
Index genesis transactions and miniblocks:
- created mint transactions
- collect delegation/stake transactions from genesis
- collect scrs from genesis
- index transaction pools (txs and scrs) and miniblocks

### CODE REVIEW

File: genesis/parsing/accountsParser.go [hunk 1]
```diff
@@ -275,7 +303,189 @@ func (ap *accountsParser) GetInitialAccountsForDelegated(addressBytes []byte) []
...
+			continue
+		}
+		tx.Signature = []byte(common.GenesisTxSignatureString)
+		tx.GasLimit = uint64(0)
+
+		txsPoolPerShard[senderShardID].Txs[string(txHash)] = tx
+		txsPoolPerShard[receiverShardID].Txs[string(txHash)] = tx
+
+		for _, miniBlock := range miniBlocks {
+			if (senderShardID == miniBlock.GetSenderShardID()) &&
```
Comment (iulianpascalau) [inline genesis/parsing/accountsParser.go:452]: could have used a bool var. The parenthesis can be removed

File: factory/processComponents.go [hunk 2]
```diff
@@ -820,9 +826,26 @@ func (pcf *processComponentsFactory) prepareGenesisBlock(genesisBlocks map[uint3
...
 }
 
-func (pcf *processComponentsFactory) indexGenesisBlocks(genesisBlocks map[uint32]data.HeaderHandler) error {
-	// In Elastic Indexer, only index the metachain block
-	genesisBlockHeader := genesisBlocks[core.MetachainShardId]
+func getGenesisBlockForShard(miniBlocks []*dataBlock.MiniBlock, shardId uint32) *dataBlock.Body {
+	var indexMiniBlocks = make([]*dataBlock.MiniBlock, 0)
+
+	for _, miniBlock := range miniBlocks {
+		if (miniBlock.GetSenderShardID() == shardId) ||
```
Comment (iulianpascalau) [inline factory/processComponents.go:833]: could have used a bool var. Useless parenthesis

File: genesis/process/intermediate/txExecutionProcessor.go [hunk 3]
```diff
@@ -53,10 +58,17 @@ func (tep *txExecutionProcessor) ExecuteTransaction(
...
 	}
 
+	tep.txs = append(tep.txs, tx)
+
 	_, err := tep.txProcessor.ProcessTransaction(tx)
 	return err
 }
 
+// GetTransactions will return the cached transactions
+func (tep *txExecutionProcessor) GetTransactions() []data.TransactionHandler {
```
Comment (iulianpascalau) [inline genesis/process/intermediate/txExecutionProcessor.go:68]: could have been named: GetExecutedTransactions

File: genesis/process/metaGenesisBlockCreator.go [hunk 4]
```diff
@@ -142,26 +155,34 @@ func createMetaGenesisBlockAfterHardFork(
...
-	return metaHdr, make([][]byte, 0), nil
+	indexingData := &genesis.IndexingData{
+		DelegationTxs:      make([]data.TransactionHandler, 0),
+		ScrsTxs:            make(map[string]data.TransactionHandler),
+		StakingTxs:         make([]data.TransactionHandler, 0),
+		DeploySystemScTxs:  make([]data.TransactionHandler, 0),
+		DeployInitialScTxs: make([]data.TransactionHandler, 0),
+	}
+
+	return metaHdr, make([][]byte, 0), indexingData, nil
```
Comment (iulianpascalau) [inline genesis/process/metaGenesisBlockCreator.go:185]: so we do not index the resulted transactions after a hardfork

File: genesis/parsing/accountsParser.go [hunk 5]
```diff
@@ -7,26 +7,40 @@ import (
...
+	minterAddressBytes []byte
+	pubkeyConverter    core.PubkeyConverter
+	keyGenerator       crypto.KeyGenerator
+	hasher             hashing.Hasher
+	marshalizer        marshal.Marshalizer
 }
 
 // NewAccountsParser creates a new decoded accounts genesis structure from json config file
 func NewAccountsParser(
 	genesisFilePath string,
```
Comment (iulianpascalau) [inline genesis/parsing/accountsParser.go:37]: could have switched to an argument DTO

File: genesis/process/metaGenesisBlockCreator.go (hunk 4 above)
Comment (AdoAdoAdo) [reply genesis/process/metaGenesisBlockCreator.go:185]: Not in this PR, as it is difficult to test at this moment.
There is some work to be done on the hardfork, but we could add a TODO

**Review by bogdan-rosianu — APPROVED** [review]

File: cmd/node/config/economics.toml [hunk 6]
```diff
@@ -16,6 +16,7 @@
          {Year = 11, MaximumInflation = 0.0},
     ]
     Denomination = 18 # represents the smallest eGLD subdivision (10^-X eGLD for a denomination of X)
+    GenesisMintingSenderAddress = "erd17rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rcqqkhty3"
```
[the remark this answers is not in the document]
Comment (ssd04) [reply cmd/node/config/economics.toml:19]: done

File: genesis/parsing/accountsParser.go [hunk 7]
```diff
@@ -275,7 +303,189 @@ func (ap *accountsParser) GetInitialAccountsForDelegated(addressBytes []byte) []
...
+	}
+
+	return txsPoolPerShard
+}
+
+func (ap *accountsParser) createMintTransactions() []coreData.TransactionHandler {
+	txs := make([]coreData.TransactionHandler, 0, len(ap.initialAccounts))
+
+	var nonce uint64 = 0
+	for _, ia := range ap.initialAccounts {
```
[the remark this answers is not in the document]
Comment (ssd04) [reply genesis/parsing/accountsParser.go]: done

File: genesis/parsing/accountsParser.go (hunk 1 above)
Comment (ssd04) [reply genesis/parsing/accountsParser.go]: done

File: factory/processComponents.go (hunk 2 above)
Comment (ssd04) [reply factory/processComponents.go]: :+1:

File: genesis/process/intermediate/txExecutionProcessor.go (hunk 3 above)
Comment (ssd04) [reply genesis/process/intermediate/txExecutionProcessor.go]: :+1:

File: genesis/parsing/accountsParser.go (hunk 5 above)
Comment (ssd04) [reply genesis/parsing/accountsParser.go]: Done

**Review by iulianpascalau — DISMISSED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**